app/components/post_body/system_message_helpers.js

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Posts} from 'mattermost-redux/constants';
+import Markdown from 'app/components/markdown';
+
+const renderUsername = (value) => {
+    return (value[0] === '@') ? value : `@${value}`;
+};
+
+const formattedMessage = (localeHolder, elements, intl) => {
```

Comment:
Just a suggestion, but you might be able to extend this function a bit to avoid repeating `return <Markdown ..../>` below instead of just simplifying the call to `intl.formatMessage`. You could do something like

```javascript
const renderMessage = (postBodyProps, styles, intl, localeHolder, values) => {
    return (
        <Markdown
            baseTextStyle={...}
            onPress={...}
            value={intl.formatMessage(localHolder, values)}
        />
    );
};
```

That would also make it easier to add any props that might be missing like the `onLongPress` prop that Elias mentioned in one of his comments.

Reply:
Done! I had originally hesitated because the function takes 5 arguments now, but not repeating is way better.